pkg/services/sqlstore/dashboard_version.go

```diff
@@ -0,0 +1,273 @@
+package sqlstore
+
+import (
+	"encoding/json"
+	"errors"
+	"time"
+
+	"github.com/grafana/grafana/pkg/bus"
+	"github.com/grafana/grafana/pkg/components/formatter"
+	"github.com/grafana/grafana/pkg/components/simplejson"
+	m "github.com/grafana/grafana/pkg/models"
+
+	diff "github.com/yudai/gojsondiff"
+	deltaFormatter "github.com/yudai/gojsondiff/formatter"
+)
+
+var (
+	// ErrUnsupportedDiffType occurs when an invalid diff type is used.
+	ErrUnsupportedDiffType = errors.New("sqlstore: unsupported diff type")
+
+	// ErrNilDiff occurs when two compared interfaces are identical.
+	ErrNilDiff = errors.New("sqlstore: diff is nil")
+)
+
+func init() {
+	bus.AddHandler("sql", CompareDashboardVersionsCommand)
+	bus.AddHandler("sql", GetDashboardVersion)
+	bus.AddHandler("sql", GetDashboardVersions)
+	bus.AddHandler("sql", RestoreDashboardVersion)
+}
+
+// CompareDashboardVersionsCommand computes the JSON diff of two versions,
+// assigning the delta of the diff to the `Delta` field.
+func CompareDashboardVersionsCommand(cmd *m.CompareDashboardVersionsCommand) error {
```

Comment:
Yeah, that makes sense. I just put it in here for convenience because it uses the unexported `getDashboardVersion` and `getDiff` helper functions. There are a few ways I could make the change. Would you prefer,
- moving it to it's own service, and copying those helper functions there?
- moving it to it's own service, exporting the helper functions here, and consuming them from the service?
- leave this function, but move the `switch` statement logic that happens on the `cmd.DiffType` to it's own service?

If there's something different you have in mind, then feel free to suggest that too!